src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.InfoCache.cs

```diff
@@ -93,9 +93,16 @@ private MemberInfo ComputeMemberInfo(IMethodSymbol sliceLikeMethod, bool require
                 // https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md#implicit-range-support
                 if (sliceLikeMethod.ReturnType.Equals(containingType))
                 {
+                    var indexer = GetIndexer(containingType, RangeType, containingType);
+                    // If the slice like method and the indexer don't both return a Ref or not, then don't
+                    // convert
+                    if (indexer != null && indexer.ReturnsByRef != sliceLikeMethod.ReturnsByRef)
+                    {
+                        return default;
```

Comment:
i don't think we want to bail out.  we'd want to perform the rest of hte if-check still right?